extensions/powershell/cmdlet-class.ts

```diff
@@ -524,14 +523,23 @@ export class CmdletClass extends Class {
 
               const codeProp = values(props).linq.first(p => p.name.toLowerCase() === 'code');
               const messageProp = values(props).linq.first(p => p.name.toLowerCase() === 'message');
+              const actionProp = values(props).linq.first(p => p.name.toLowerCase() === 'action');
 
               if (codeProp && messageProp) {
-                const lcode = new LocalVariable('code', dotnet.Var, { initializer: `(await response).${ep}${codeProp.name};` });
-                const lmessage = new LocalVariable('message', dotnet.Var, { initializer: `(await response).${ep}${messageProp.name};` });
+                const lcode = new LocalVariable('code', dotnet.Var, { initializer: `(await response)?.${ep}${codeProp.name}` });
```

Comment:
The newly added question mark in these three lines is crucial to avoid a NullReferenceException when the error response does not contain the expected object. This can occur because the error was generated (often with an empty body) by a layer in front of the API being called.

Reply:
The action handling is nice to have, but fixing the nullref is critical.